File: apps/nex-force/src/types/asset.ts
/**
 * Asset types — converted from Angular's assets.service.ts interfaces.
 *
 * Changes:
 *   - Removed duplicate `AssetDTO` (merged with `AssetData`)
 *   - Made field names consistent (camelCase)
 *   - Used `string` for dates (API returns ISO strings)
 */

export interface Asset {
    assetId: number;
    assetName: string;
    assetModel: string;
    assetCode: string;
    productNo: string;
    type: string;
    serialNumber: string;
    brand: string;
    cost: number;
    location: string;
    warrantyStart: string;
    warrantyEnd: string;
    warranty: string;
    vendor: string;
    category: string;
    condition: string;
    description: string;
    status: string;
    isActive: boolean;
    employeeId: number | null;
    assignedDate: string;
    assetUser: string;
    assetImg1: string | null;
    assetImg2: string | null;
    assetImg3: string | null;
    assetImg4: string | null;
    createDate: string | null;
    createBy: string | null;
    updateDate: string | null;
    updateBy: string | null;
}

export interface AssetData {
    id: number;
    assetId: number;
    assetName: string;
    assetCode: string;
    assignedDate: string;
    category: string;
    type: string;
    serialNumber: string;
    brand: string;
    cost: number;
    location: string;
    warrantyStart: string;
    warrantyEnd: string;
    warranty: string;
    vendor: string;
    assetModel: string;
    productNo: string;
    assetImg1?: string;
    assetImg2?: string;
    assetImg3?: string;
    assetImg4?: string;
    assetUser: string;
    description: string;
    status: string;
}

export interface AssetResponse {
    data: AssetData[];
    totalData: number;
}

export type AssetStatus = "Pending" | "Approved" | "Deployed" | "Returned";

export interface AssigneeData {
    name: string;
    profileImg: string;
    email: string;
    id: number;
}

File: services/nex-force-api/Middleware/Models/Asset.cs
using Middleware.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Middlewares.Models
{
    [Table("adm-tb-ms-assets")]
    public class Asset
    {
        public int AssetId { get; set; }                     // Auto-incremented asset ID
        public string? AssetName { get; set; }                // Name of the asset (e.g., Dell Keyboard)
        public string? AssetModel { get; set; }
        public string? ProductNo { get; set; }
        public string? Type { get; set; }                     // Type of the asset (e.g., Wired Keyboard)
        public string? SerialNumber { get; set; }             // Serial number of the asset
        public string? Brand { get; set; }                    // Brand of the asset (e.g., Dell)
        public decimal? Cost { get; set; }                    // Cost of the asset
        public string? Location { get; set; }                 // Location of the asset
        public DateTime? WarrantyStart { get; set; }          // Warranty start date
        public DateTime? WarrantyEnd { get; set; }            // Warranty end date
        public string? Vendor { get; set; }                   // Vendor name
        public string? Category { get; set; }                 // Asset category
        public bool IsActive { get; set; }                    // Active status
        public DateTime? CreateDate { get; set; }             // Creation date
        public string? CreateBy { get; set; }                 // Creator's name
        public DateTime? UpdateDate { get; set; }             // Update date
        public string? UpdateBy { get; set; }
        public string? AssetImg1 { get; set; }
        public string? AssetImg2 { get; set; }
        public string? AssetImg3 { get; set; }
        public string? AssetImg4 { get; set; }
        public decimal? EmployeeId { get; set; }
        public DateTime? AssignedDate	 { get; set; }
        public string? AssetCode { get; set; }
        public string? Supplier { get; set; }
        public decimal? Warranty { get; set; }
        public string? Description { get; set; }
        public string? Status { get; set; }
        public string? Condition { get; set; }
        public virtual Employee? Employee { get; set; } // Navigation property
    }
}

